### PR TITLE
blog post for MatchLabelKeys in PodAffinity

### PR DESCRIPTION
**Pull request by sanposhiho**
~This is just a place holder PR for now.~

It's a blog post for https://github.com/kubernetes/enhancements/issues/3633

### CODE REVIEW

**Comment by sftim** [comment]
/hold

pending assignment of publication date (by release comms)

we'll also keep this held until the v1.31 release has happened

**Comment by mbianchidev** [comment]
Hey @sanposhiho , it's Matteo from the 1.31 Release comms team. 
I'm reaching out with a reminder that the blog **ready for review deadline is July 26, 2024**. 

Please let me know how we can help with content, review or anything else 😄

**Comment by sanposhiho** [comment]
Now it's ready for review.

**Comment by mbianchidev** [comment]
Blog publish date is currently scheduled for August 16, 2024 @sanposhiho 
I'll take a look this or next week for a review :)

**Comment by sanposhiho** [comment]
@Huang-Wei can you also possibly take a look from KEP reviewer pov?

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:14]: `accurate` doesn't sound accurate lol... It's more like adding a new feature/knob to adapt to wider/more practical usecases. Can we use a better word here?

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:16]:
```suggestion
During a workload's (e.g., Deployment) rolling upgrade, a cluster may have Pods from multiple versions at the same time. 
```

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:17]:
```suggestion
However, the scheduler cannot distinguish between old and new versions based on the `LabelSelector` specified in PodAffinity or PodAntiAffinity. As a result, it will co-locate or disperse Pods regardless of their versions.
```

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:19]:
```suggestion
This can lead to sub-optimal scheduling outcome, for example:
```

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:20]:
```suggestion
- New version Pods are co-located with old version Pods (PodAffinity), which will eventually be removed after rolling upgrades.
```

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [inline content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:21]:
```suggestion
- Old version Pods are distributed across all available topologies, preventing new version Pods from finding nodes due to PodAntiAffinity.
```

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (Huang-Wei) [reply content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md:25]: +1

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (sanposhiho) [reply content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md]: > ## `MatchLabelKeys` - Enabling version-aware scheduling for rolling upgrades

How does it sound now?

Comment (Huang-Wei) [reply content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md]: How about:

> `MatchLabelKeys` - enhanced scheduling for versatile rolling upgrades

**Review by mbianchidev — APPROVED** [review]
My review is done and this is a content-wise approve!
+1 from me

File: content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md
Comment (sanposhiho) [reply content/en/blog/_posts/2024-xx-xx-matchlabelkeys-podaffinity.md]: Done.

**Comment by tengqm** [comment]
@sanposhiho You may want to change "rolling upgrade" to "rolling update" if possible. Updating a Deployment doesn't always translate to "upgrade"; it could be a "downgrade" in some cases.

**Comment by sanposhiho** [comment]
Good point, fixed.

**Comment by T-Lakshmi** [comment]
[Preview page](https://deploy-preview-46833--kubernetes-io-main-staging.netlify.app/blog/2024/08/16/matchlabelkeys-podaffinity/)

**Comment by mbianchidev** [comment]
@natalisucks this one is approved and requires a lgtm label :eyes:

**Comment by natalisucks** [comment]
/hold
/lgtm

Copying @sftim's warning note, thanks Tim!

### 🛑 Do not unhold or merge before Kubernetes v1.31 is released 🛑

**Comment by a-mccarthy** [comment]
/unhold

Kubernetes v1.31 has been released.

**Pull request merged**
